datadog/fwprovider/data_source_datadog_app_builder_app.go

```diff
@@ -0,0 +1,195 @@
+package fwprovider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var _ datasource.DataSource = &appBuilderAppDataSource{}
+
+type appBuilderAppDataSource struct {
+	Api  *datadogV2.AppBuilderApi
+	Auth context.Context
+}
+
+func NewDatadogAppBuilderAppDataSource() datasource.DataSource {
+	return &appBuilderAppDataSource{}
+}
+
+func (d *appBuilderAppDataSource) Configure(_ context.Context, request datasource.ConfigureRequest, response *datasource.ConfigureResponse) {
+	providerData := request.ProviderData.(*FrameworkProvider)
+	d.Api = providerData.DatadogApiInstances.GetAppBuilderApiV2()
+	d.Auth = providerData.Auth
+}
+
+func (d *appBuilderAppDataSource) Metadata(_ context.Context, request datasource.MetadataRequest, response *datasource.MetadataResponse) {
+	response.TypeName = "app_builder_app"
+}
+
+func (d *appBuilderAppDataSource) Schema(_ context.Context, request datasource.SchemaRequest, response *datasource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Description: "This data source retrieves the definition of an existing Datadog App from App Builder for use in other resources, such as embedding Apps in Dashboards or Notebooks.",
```

Comment:
```suggestion
		Description: "This data source retrieves the definition of an existing Datadog App from App Builder for use in other resources, such as embedding Apps in Dashboards.",
```
Notebooks doesn't exist as a Terraform resource.
As for dashboards, I'm ok with this example mentioned but customers will only be able to embed it when using the [dashboard JSON-based resource](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/dashboard_json), since the non-JSON one doesn't seem to support this right now.
I'm not sure this data source is extremely useful right now, but it could be kept for future integrations.
As long as the change in description is accepted, I'm happy to approve.